### ISSUE TITLE
mean_threshold parameter does not impact output

### ISSUE THREAD

**Issue by nalinratnayeke** [instanseg/instanseg#136]
Thanks for putting together this package! I've been excited to integrate it into our pipeline but in the process of optimizing segmentation parameters, I noticed that mean_threshold does not appear to impact the segmentation instances. I'm interested in using mean_threshold to help filter out some low probability spurious cells that are identified.

I've replicated the issue using the segmentation_minimal.ipynb example fluorescence image and fluorescence model just changing mean_threshold to various values ranging from 0 to 1 to even higher like 100 and 1e6.

```
image_array, pixel_size = instanseg_fluorescence.read_image("../instanseg/examples/LuCa1.tif")

labeled_output, image_tensor = instanseg_fluorescence.eval_small_image(
    image_array, pixel_size,
    resolve_cell_and_nucleus=True,
    cleanup_fragments=True,
    mean_threshold=0.3
)

display = instanseg_fluorescence.display(image_tensor, labeled_output)
show_images(display, colorbar=False)
```

I'm not sure if I'm misunderstanding what mean_threshold is supposed to be doing (basing it off of this related issue https://github.com/instanseg/instanseg/issues/29) , but it would be great if you could provide some clarification here. Thanks in advance!

**Comment by nalinratnayeke** [comment]
Hi @ThibautGoldsborough, just bumping this to see if this is expected behavior, thanks very much!

**Comment by ThibautGoldsborough** [comment]
Hi @nalinratnayeke . 

Thanks for raising this,

The short answer is that `mean_threshold` is an artifact here, apologies.
 
`mean_threshold` is a threshold of mean seed map under a detected object. It make sense to use it when the seed loss function is cross entropy, as the average seed value is related to the probability of an object existing. For `instanseg_fluorescence`, the seed map predicts a distance map to the object boundary (bot a binary map), so the mean seed map value under an object is much harder to interpret, and may not actually relate to a probability of an object existing but more the shape and convexity of the object, which is why I switched it off for those models.

So this value is an artifact for the public models, `seed_threshold` is the parameter that you should use instead, apologies for the confusion.

**Comment by nalinratnayeke** [comment]
Thanks very much for the response! Yes, seed_threshold is working well for tuning the segmentation.
